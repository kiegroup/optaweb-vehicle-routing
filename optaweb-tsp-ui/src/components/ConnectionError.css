.modal-container {
    position: relative;
  }
  .modal-container .modal, .modal-container .modal-backdrop {
    position: absolute;
  }
