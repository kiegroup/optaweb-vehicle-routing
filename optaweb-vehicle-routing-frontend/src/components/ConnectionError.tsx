/*
 * Copyright 2018 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { Modal, Text, TextContent, TextVariants } from '@patternfly/react-core';
import * as React from 'react';

export interface IConnectionErrorProps {
  title: string;
  message: string;
  icon?: React.ReactNode;
  help?: string;
}

const renderHelpBlock = (help?: string) => {
  return help ? (
    <Text component={TextVariants.small}>{help}</Text>
  ) : (
    ''
  );
};

const ConnectionError: React.SFC<IConnectionErrorProps> = (
  { title, message, icon, help }: IConnectionErrorProps,
) => {
  return (
    <Modal title={title} isOpen={true}>
      <TextContent>
        <Text component={TextVariants.h3}>
          {icon}
          {message}
          {renderHelpBlock(help)}
        </Text>
      </TextContent>
    </Modal>
  );
};

export default ConnectionError;
