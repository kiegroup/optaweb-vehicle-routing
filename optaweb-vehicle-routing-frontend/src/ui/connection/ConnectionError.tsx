/*
 * Copyright 2018 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { Modal, Text, TextContent, TextVariants } from '@patternfly/react-core';
import * as React from 'react';

export interface ConnectionErrorProps {
  title: string;
  message: string;
  icon?: React.ReactNode;
  help?: string;
  isOpen: boolean;
}

const renderHelpBlock = (help?: string) => (help ? <Text component={TextVariants.small}>{help}</Text> : '');

const ConnectionError: React.FC<ConnectionErrorProps> = ({
  title,
  message,
  icon,
  help,
  isOpen,
}) => (
  <Modal title={title} isOpen={isOpen} isSmall>
    <TextContent>
      <Text component={TextVariants.h3}>
        {icon}
        {message}
        {renderHelpBlock(help)}
      </Text>
    </TextContent>
  </Modal>
);

export default ConnectionError;
