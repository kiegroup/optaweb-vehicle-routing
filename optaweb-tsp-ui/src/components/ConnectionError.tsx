import { Modal, Text, TextContent, TextVariants } from '@patternfly/react-core';
import * as React from 'react';
import { ReactNode } from 'react';
import './ConnectionError.css';
export interface IConnectionErrorProps {
  title: string;
  message: string;
  icon?: ReactNode;
  help?: string;
}

export default class ConnectionError extends React.Component<
  IConnectionErrorProps
> {
  constructor(props: IConnectionErrorProps) {
    super(props);
  }

  renderHelpBlock() {
    const { help } = this.props;
    return help ? (
      <Text component={TextVariants.small}>{this.props.help}</Text>
    ) : (
      ''
    );
  }

  render() {
    const { title, message, icon } = this.props;
    return (
      <Modal title={title} isOpen={true} style={{ zIndex: 1000000 }}>
        <TextContent>
          <Text component={TextVariants.h3}>
            {icon}
            {message}
            {this.renderHelpBlock()}
          </Text>
        </TextContent>
      </Modal>
    );
  }
}
