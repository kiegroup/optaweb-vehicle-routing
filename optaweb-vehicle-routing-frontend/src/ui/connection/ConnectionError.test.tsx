/*
 * Copyright 2018 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import * as React from 'react';
import ConnectionError, { IConnectionErrorProps } from './ConnectionError';

describe('Connection Error Component', () => {
  it('should render correctly', () => {
    const props: IConnectionErrorProps = {
      title: 'title',
      message: 'message',
      icon: <i>Icon</i>,
      help: 'help',
      isOpen: true,
    };

    expect.assertions(1);
    const connectionError = shallow(<ConnectionError {...props} />);
    expect(toJson(connectionError)).toMatchSnapshot();
  });
});
