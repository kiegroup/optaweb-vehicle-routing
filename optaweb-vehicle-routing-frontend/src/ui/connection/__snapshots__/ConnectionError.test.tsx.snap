// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connection Error Component should render correctly 1`] = `
<Modal
  actions={Array []}
  ariaDescribedById=""
  className=""
  hideTitle={false}
  isLarge={false}
  isOpen={true}
  isSmall={true}
  onClose={[Function]}
  title="title"
>
  <TextContent>
    <Text
      component="h3"
    >
      <i>
        Icon
      </i>
      message
      <Text
        component="small"
      >
        help
      </Text>
    </Text>
  </TextContent>
</Modal>
`;
